### PR TITLE
Plugin Upload: show notice outside of data-layer AT status polling code

### PR DESCRIPTION
**Pull request by lamosty**
Currently, we show a success notice "You've successfully uploaded the %(pluginId)s plugin." after polling transfer status and getting a `complete` status. However, the code which is used to poll for a transfer status can be used outside of just the Plugin Upload page (where this notice makes sense). 

In this PR, we are moving the notice away from the data-layer code and into the Plugin Upload component where it makes sense.

## Testing

Select a Business site and navigate to `/plugins/upload/<site>`. Upload a plugin zip. After the site transfers, you should be redirected to the plugin info view and the notice should show.

### CODE REVIEW

**Comment by seear** [comment]
This component [also does jetpack plugin uploads](https://github.com/Automattic/wp-calypso/blob/7d071df04eb2a0755f41d960cfc9b85e17112ff4/client/my-sites/plugins/plugin-upload/index.jsx#L87). For these, the success notice is handled by the plugins own flux events, triggered [here](https://github.com/Automattic/wp-calypso/blob/112c7f3c62e8c089dd2756fe40de99fb90e50b4e/client/state/data-layer/wpcom/sites/plugins/new/index.js#L74). This is the result of uploading a plugin to a jetpack site on this branch:

screen shot 2017-10-05 at 13 44 32


We probably want to only display the success notice from the component if the (original) site was not jetpack, or if the current site is atomic. This could be tricky given that we are not too sure when the transition occurs for transfers?

**Comment by seear** [comment]
> the success notice is handled by the plugins section's own flux events, triggered here.

Oh, also, this flux dispatch is necessary because it adds the plugin data to the flux store so we can show the plugin page. The success notice comes for free and it is easier to use it than not show it.

**Comment by lamosty** [comment]
Good catch. I think I got a solution in 26be96882a0bfb9d2d8edff6c12efe7a1d6b6e36.

File: client/my-sites/plugins/plugin-upload/index.jsx [hunk 1]
```diff
@@ -63,6 +69,20 @@ class PluginUpload extends React.Component {
 		if ( nextProps.complete ) {
 			page( `/plugins/${ nextProps.pluginId }/${ nextProps.siteSlug }` );
 		}
+
+		const { COMPLETE } = transferStates;
+
+		if (
+			this.props.automatedTransferStatus !== COMPLETE &&
+			nextProps.automatedTransferStatus === COMPLETE
```
Comment (seear) [inline client/my-sites/plugins/plugin-upload/index.jsx]: Nice solution 👍

**Pull request merged**